docs: Updates to the Project Structure article

ggarabedian opened the pull request:
Addresses #1257 and #1360.

- [x] Modify the document to explain the Angular project structure as well.
- [x] Clean up the Android Runtime Custom Flags article.

Looking for comments, suggestions and ideas on how to improve the article. :)

tjvantoll commented on docs/core-concepts/android-runtime/advanced-topics/custom-flags.md:
```diff
@@ -1,57 +1,56 @@
 ---
-nav-title: "Custom Flags"
 title: "Custom Flags"
 description: "Tuning with custom flags"
```
This didn’t change, but while you’re in this article, if you have a minute to give this page a better description that would be cool.

tjvantoll commented on docs/core-concepts/android-runtime/advanced-topics/custom-flags.md:
```diff
@@ -1,57 +1,56 @@
+# Custom Flags
 
-## Tuning V8
+The V8 engine comes with a set of controlling flags that may be useful for a fine-grained tuning. You can set these flags in the [secondary package.json configuration file]({% slug structure %}#apppackagejson). This article contains some of the available flags and short explanation on how you can use them. For a complete list of all V8 flags, see the [Flag Definitions header file](https://github. ...
```
```suggestion
The V8 engine comes with a set of controlling flags that may be useful for fine-grained tuning. You can set these flags in the [secondary package.json configuration file]({% slug structure %}#apppackagejson). This article contains some of the available flags and a short explanation on how you can use them. For a complete list of all V8 flags, see the [Flag Definitions header file](https://github.com/v8/v8/blob/6.9.247/src/flag-definitions.h) in GitHub.
```

tjvantoll replied:
Also, I totally didn’t realize you could do `{% slug page-name}#heading` and I’ve needed it a few times 🙂

ggarabedian replied:
Yeah, it's very useful when you try to link to an anchor on another page. If later the link changes, it doesn't break as much as when it is hardcoded :)

tjvantoll commented on docs/core-concepts/android-runtime/advanced-topics/custom-flags.md:
```diff
@@ -1,57 +1,56 @@
-For improved performance V8 keeps a cache of various values used for date / time computation. This may lead to a negative side effect for the application because changes made to the current timezone will not be reflected until the application is restarted.
-
-While this is not a common requirement for most applications, under some circumstances this be needed. To enable this scenario, you can set the `handleTimeZoneChanges` flag:
+For improved performance V8 keeps a cache of various values used for date and time computation. This may lead to a negative side effect for the application because changes made to the current timezone will not be reflected until the application is restarted. While this is not a common requirement for most applications, there are some circumstances where this behavior might be needed. To enable th ...
```
```suggestion
For improved performance, V8 keeps a cache of various values used for date and time computation. This may lead to a negative side effect for the application, because changes made to the current timezone will not be reflected until the application is restarted. While this is not a common requirement for most applications, there are some circumstances where this behavior might be needed. To enable this scenario, you can set the `handleTimeZoneChanges` flag:
```

tjvantoll commented on docs/core-concepts/android-runtime/advanced-topics/custom-flags.md:
```diff
@@ -75,11 +74,10 @@ By default, if an exception is thrown when executing JavaScript code which is ca
 
-This flag works for iOS as well.
-
+> **Note:** The `discardUncaughtJsExceptions` works on iOS as well. This is why in the example above the flag is not assigned to the `android` object.
```
```suggestion
> **NOTE:** The `discardUncaughtJsExceptions` works on iOS as well. This is why in the example above the flag is not assigned to the `android` object.
```

tjvantoll replied:
Trivial, but we usually capitalize these.

tjvantoll commented on docs/core-concepts/android-runtime/advanced-topics/memory-management.md:
```diff
@@ -1,8 +1,8 @@
 ---
-nav-title: "Memory Management"
 title: "Memory Management"
 description: "Advanced Memory Management"
```
Same meh description here. Not a huge deal.

ggarabedian replied:
Done :)

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+{% endangular%} 
+
+{% nativescript %}
+{% endnativescript %}
```
?

ggarabedian replied:
I am not sure why this one, the "TODOs" and the missing closing brackets are still present in this PR. They are not in my local repo and I am sure that I pushed. Anyway, thanks for pointing them out and sorry :))

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
 
-The default structure of a NativeScript project consist of a project directory that contains: `app`, `platforms`, `node_modules` directories and the `package.json` of the project. The `app` folder contains the source of the application as well as the `App_Resources` folder. `App_Resources` folder contains the platform-specific resources of the application (icons, configuration files etc.).
+{% nativescript %}
+The default structure of a blank NativeScript project consist of a root folder that contains the `app`, `platforms` and `node_modules` directories, and a `package.json` configuration file. 
```
```suggestion
The default structure of a blank NativeScript project consists of a root folder that contains the `app`, `platforms` and `node_modules` directories, and a `package.json` configuration file. 
```

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+{% endnativescript %}
+
+{% angular %}
+The default structure of a blank NativeScript + Angular project consist of a root folder that contains the `src`, `platforms`, `node_modules` and `hooks` directories, and several configuration files amongs which the most important is the `package.json`. 
```
```suggestion
The default structure of a blank NativeScript + Angular project consists of a root folder that contains the `src`, `platforms`, `node_modules` and `hooks` directories, and several configuration files—amongst which the most important is the `package.json`. 
```

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+##{% nativescript %}# app/{% endnativescript %}App_Resources
 
-The `platforms` directory is created when you start a build or add a target platform to your project, the NativeScript tooling creates a new subdirectory with the platform name. These subdirectories have the platform-specific project structure required for native development with the native SDKs for the platform. When the project is prepared for build, the NativeScript tooling copies relevant con ...
+The `App_Resources` folder contains the platform-specific resources of the application (icons, configuration files etc.). The configuration files that are respected by NativeScript tooling are respectively the `App_Resources/Android/src/main/AndroidManifest.xml` for Android and the `App_Resources/iOS/Info.plist` for iOS.
```
```suggestion
The `App_Resources` folder contains the platform-specific resources of the application (icons, configuration files etc.). The configuration files that are respected by NativeScript tooling are the `App_Resources/Android/src/main/AndroidManifest.xml` for Android, and the `App_Resources/iOS/Info.plist` for iOS.
```

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+The `App_Resources` folder contains the platform-specific resources of the application (icons, configuration files etc.). The configuration files that are respected by NativeScript tooling are respectively the `App_Resources/Android/src/main/AndroidManifest.xml` for Android and the `App_Resources/iOS/Info.plist` for iOS.
 
-**IMPORTANT:** Avoid editing files located in the `platforms` subdirectory because the NativeScript CLI overrides such files during the `prepare <Platform>` using the contents of the `app` directory.
+(!!!!! FIX PATHS TO FILES !!!!)
```
:)

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+## The **platforms** directory
 
-The package.json file also contains information about npm packages (including [NativeScript plugins]({%slug plugins-infrastructure %})) inside `dependencies` and `devDependencies` keys.
+The `platforms` directory is created when you start a build or add a target platform to your project. The NativeScript tooling creates a new subdirectory with the respective platform name. These subdirectories have the platform-specific project structure required for native development with the native SDKs of the platform. When the project is prepared for build, the NativeScript tooling copies re ...
```
```suggestion
The `platforms` directory is created when you start a build or add a target platform to your project. The NativeScript tooling creates a new subdirectory with the respective platform name. These subdirectories have the platform-specific project structure required for native development with the native SDKs of the platform. When the project is prepared for build, the NativeScript tooling copies relevant content from the {% nativescript %}`app`{% endnativescript %}{% angular %}`src`{% endangular%} directory to the platform-specific subdirectory for each target platform.
```

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+The `platforms` directory is created when you start a build or add a target platform to your project. The NativeScript tooling creates a new subdirectory with the respective platform name. These subdirectories have the platform-specific project structure required for native development with the native SDKs of the platform. When the project is prepared for build, the NativeScript tooling copies re ...
 
-## **nsconfig.json** file
+> **IMPORTANT:** Avoid editing files located in the `platforms` subdirectories because the NativeScript CLI overrides them with the content of the % nativescript %}`app`{% endnativescript %}{% angular %}`src`{% endangular%} directory during the `prepare <Platform>` process.
```
```suggestion
> **IMPORTANT:** Avoid editing files located in the `platforms` subdirectories because the NativeScript CLI overrides them with the content of the {% nativescript %}`app`{% endnativescript %}{% angular %}`src`{% endangular%} directory during the `prepare <Platform>` process.
```

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+
+The root `package.json` also contains several NativeScript-specific properties which are placed inside the `nativescript` object:
+
+* **id** - Specifies the unique application identifier (App ID) of the app. To be able to build for both Android and iOS, your App ID must be unique and contain two or more strings, separated by a dot. Each string must start with a letter and should contain only letters and numbers. The app identifier must not start with an uppercase letter. (!!!!! LINK DOCUMENT ABOUT SEPARATE IDENTIFIERS -> NS 5 ...
```
Just fyi you have another placeholder in here.

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+    "nativescript": {
+        "id": "org.nativescript.myApplication",
+        "tns-android": {
+            "version": "4.2.0"
```
Probably worth worth making all of these 5.0.

tjvantoll commented on docs/core-concepts/project-structure.md:
```diff
@@ -4,100 +4,187 @@ description: Learn the basic project structure of а NativeScript application -
+
+## The **tsconfig.json** file
+
+The `tsconfig.json` file is present only in projects that use TypeScript and its function is to work as a guide during the [transpilation]({% slug transpilers %}) of TypeScript to JavaScript. You can fine-tune the transpilation process by configuring the various [compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html). For more information about `tsconfig.json`, see  ...
```
```suggestion
The `tsconfig.json` file is present only in projects that use TypeScript. The file works as a guide during the [transpilation]({% slug transpilers %}) of TypeScript to JavaScript. You can fine-tune the transpilation process by configuring the various [compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html). For more information about `tsconfig.json`, see the official [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html).
```

ggarabedian commented:
Thank you for the feedback @tjvantoll! I addressed all of your comments. You might want to check the descriptions of the Custom Flags and Memory Management articles, and let me know if they seem okay to you

tjvantoll commented on docs/core-concepts/android-runtime/advanced-topics/custom-flags.md:
```diff
@@ -1,57 +1,56 @@
-nav-title: "Custom Flags"
 title: "Custom Flags"
-description: "Tuning with custom flags"
+description: "Configure various V8 engine flags in order to improve the performance of your app or obtain more comprehensive information during debugging"
```
```suggestion
description: "Configure various V8 engine flags in order to improve the performance of your app, or to obtain more comprehensive information during debugging"
```
